src/main/java/htsjdk/samtools/cram/structure/block/Block.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * ****************************************************************************
+ * Copyright 2013 EMBL-EBI
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ****************************************************************************
+ */
+package htsjdk.samtools.cram.structure.block;
+
+import htsjdk.samtools.cram.CRAMException;
+import htsjdk.samtools.cram.common.CramVersions;
+import htsjdk.samtools.cram.compression.ExternalCompression;
+import htsjdk.samtools.cram.compression.ExternalCompressor;
+import htsjdk.samtools.cram.io.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+/**
+ * Class representing CRAM block concept and some methods to operate with block content. CRAM block is used to hold some (usually
+ * homogeneous) binary data. An external compression can be applied to the content of a block. The class provides some instantiation static
+ * methods, for example to read a block from an input stream. Blocks can be written out to an output stream, this may be considered as a way
+ * to serialize/deserialize blocks.
+ */
+public abstract class Block {
+    /**
+     * Only external blocks have meaningful Content IDs
+     * Other blocks are required to have a Content ID of 0
+     */
+    public static final int NO_CONTENT_ID = 0;
+
+    /**
+     * Identifies CRAM content type of the block.
+     */
+    private final BlockContentType contentType;
+
```

Comment:
Would it make sense to have the BlockCompressionMethod kept here as a final, and just have `RawBlock` enforce it to always be RAW ? Same question about rawContent (since both subclasses have that). Then this can have default implementations for everything; `RawBlock` would be simplified, and `CompressibleBlock` would only override the compressedContent part.

Reply:
Yes - and we do not actually need `CompressibleBlock`